### PR TITLE
sharing into Keybase for both chat and Files on iOS

### PR DESCRIPTION
**Pull request by songgao**
Screencast: [keybase://team/keybase/videos/sharing-into-keybase-ios-20200210.mov](keybase://team/keybase/videos/sharing-into-keybase-ios-20200210.mov)

Things that work:

* share multiple pictures or a single picture
* share PDF(s)
* share into chat
* share into KBFS

Things not working but I'd like to get to work in the future:

* [x] fix sharing URL (iOS)
* [x] fix sharing text and URL (iOS)
* [x] fix the issue where when sending image(s), the content field of the native popup isn't sent (iOS)
* [ ] error handling (iOS)
* [x] text sharing on Android

### CODE REVIEW

**Comment by mmaxim** [comment]
Think we need the text share stuff to work, but I will look at this.

**Comment by mmaxim** [comment]
Also why does it have a send button on the chat modal? Should just go if you tap it. I am refactoring that component now anyway, so maybe I can just look at it.

**Comment by songgao** [comment]
> Think we need the text share stuff to work, but I will look at this.

Agree; I made a polish ticket to work on this next sprint.

> Also why does it have a send button on the chat modal? Should just go if you tap it. I am refactoring that component now anyway, so maybe I can just look at it.

That sounds reasonable, although that's from [design](https://zpl.io/VQ9Bjv5) but I can't remember the "send" button was necessary, so perhaps check with them?

**Comment by mmaxim** [comment]
I blew it away in my upcoming PR

**Comment by mmaxim** [comment]
Is there a way this can be feature flagged until the rest is done?

**Comment by songgao** [comment]
I deleted a bunch of code from the native extension so probably not easily. But I could continue iterate on this branch until it's ready

**Comment by songgao** [comment]
(not ready yet; will get to Android tomorrow)

**Comment by songgao** [comment]
@chrisnojima @mmaxim this is ready

File: shared/actions/fs/index.tsx [hunk 1]
```diff
@@ -586,33 +586,60 @@ const moveOrCopy = async (state: Container.TypedState, action: FsGen.MovePayload
...
-    } else {
-      await RPCTypes.SimpleFSSimpleFSCopyRecursiveRpcPromise(params)
-    }
-    return RPCTypes.SimpleFSSimpleFSWaitRpcPromise({opID: params.opID})
+    const rpc =
+      action.type === FsGen.move
+        ? RPCTypes.SimpleFSSimpleFSMoveRpcPromise
+        : RPCTypes.SimpleFSSimpleFSCopyRecursiveRpcPromise
+    await Promise.all(params.map(p => rpc(p)))
+    return Promise.all(params.map(({opID}) => RPCTypes.SimpleFSSimpleFSWaitRpcPromise({opID})))
```
[the remark this answers is not in the document]
Comment (songgao) [reply shared/actions/fs/index.tsx]: Hmm not sure what you mean. Could you elaborate? Those calls are async and this waits for them to finish.

[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/actions/fs/index.tsx]: If one errors they all stop

[the remark this answers is not in the document]
Comment (songgao) [reply shared/actions/fs/index.tsx]: Ah it's on Promise. Cool thanks

**Comment by mmaxim** [comment]
I find the UX of the native window popping up with no selection option kind of strange. It looks bugged out, how am I supposed to know what it's going to do when I hit "Post"? As it is now, it looks like it will just send to nowhere. 

Also I put a build of this on my phone and it didn't work. I hit post and it doesn't open Keybase.

**Comment by songgao** [comment]
> I find the UX of the native window popping up with no selection option kind of strange. It looks bugged out, how am I supposed to know what it's going to do when I hit "Post"? As it is now, it looks like it will just send to nowhere.

I was thinking we could skip the post popup but just show a "preparing" popup then open Keybase, but didn't actually do that in this PR. I guess that should be included.

**Comment by songgao** [comment]
> Also I put a build of this on my phone and it didn't work. I hit post and it doesn't open Keybase.

Reproed this using a large image from the Photos app and I saw a bunch of memory warnings in the log. I think it's because I used PNG representation when saving the image and it went over the limit. Changing to JPG fixes it. I'll update this branch soon.

**Comment by songgao** [comment]
@mmaxim ready again!

**Comment by mmaxim** [comment]
Think it's almost there, but sharing multiple photos causes the app to not open up.

**Comment by songgao** [comment]
got chat thumb; merging!

**Pull request merged**